sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryComposeHelper.kt

```diff
@@ -0,0 +1,91 @@
+@file:Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE") // to access internal vals
+
+package io.sentry.compose
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.node.LayoutNode
+import androidx.compose.ui.node.LayoutNodeLayoutDelegate
+import androidx.compose.ui.semantics.SemanticsModifier
+import io.sentry.ILogger
+import io.sentry.SentryLevel
+import java.lang.reflect.Field
+
+internal class SentryComposeHelper(private val logger: ILogger) {
+
+    private val layoutDelegateField: Field? =
+        loadField(logger, "androidx.compose.ui.node.LayoutNode", "layoutDelegate")
+    private val testTagElementField: Field? =
+        loadField(logger, "androidx.compose.ui.platform.TestTagElement", "tag")
+    private val sentryTagElementField: Field? =
+        loadField(logger, "io.sentry.compose.SentryModifier.SentryTagModifierNodeElement", "tag")
+
+    fun extractTag(modifier: Modifier): String? {
+        val type = modifier.javaClass.canonicalName
+        // Newer Jetpack Compose uses TestTagElement as node elements
+        // See
+        // https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/platform/TestTag.kt;l=34;drc=dcaa116fbfda77e64a319e1668056ce3b032469f
+        try {
+            if ("androidx.compose.ui.platform.TestTagElement" == type &&
+                testTagElementField != null
+            ) {
+                val value = testTagElementField.get(modifier)
+                return value as String?
+            } else if ("io.sentry.compose.SentryModifier.SentryTagModifierNodeElement" == type &&
+                sentryTagElementField != null
+            ) {
+                val value = sentryTagElementField.get(modifier)
+                return value as String?
+            }
+        } catch (e: Throwable) {
+            // ignored
+        }
+
+        // Older versions use SemanticsModifier
+        if (modifier is SemanticsModifier) {
+            val semanticsConfiguration =
+                modifier.semanticsConfiguration
+            for ((item, value) in semanticsConfiguration) {
+                val key = item.name
+                if ("SentryTag" == key || "TestTag" == key) {
+                    if (value is String) {
+                        return value
+                    }
+                }
+            }
+        }
+        return null
+    }
+
+    fun getLayoutNodeBoundsInWindow(node: LayoutNode): Rect? {
+        if (layoutDelegateField != null) {
+            try {
+                val delegate =
```

Comment:
hmm, I'm thinking you could directly access coordinates in LayoutNode `node.coordinates.boundsInWindow()` without reflection. At least that's what I'm doing in here: https://github.com/getsentry/sentry-java/blob/801e6773af3b3543214ddd2453264fc5a9c85f00/sentry-android-replay/src/main/java/io/sentry/android/replay/viewhierarchy/ComposeViewHierarchyNode.kt#L85

however I think it returns an [innerCoordinator](https://github.com/androidx/androidx/blob/ad593cc5efcc0703fd177d5a806c0a8898ba914a/compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/node/LayoutNode.kt#L984-L985) as opposed to outerCoordinator you use - do you know the difference between those? 😅

Reply:
Also, I think you could make use of this vendored `boundsInWindow` method too right? https://github.com/getsentry/sentry-java/blob/801e6773af3b3543214ddd2453264fc5a9c85f00/sentry-android-replay/src/main/java/io/sentry/android/replay/util/Nodes.kt#L161-L206

You also have access to the root node (at least in the view hierarchy exporter), I think it would speed up things by quite a bit. Ideally we'd share that method via a separate module, but for simplicity I'd suggest just copying it over in this package for now to not deal with another module.

Reply:
Since the ViewHierarchy uses relative coordinates, I switched to `boundsInParent()` which is looking good: https://sentry-sdks.sentry.io/issues/5893003489/events/bd2a88e3d9a9436eb3f93004753d9715?project=5428559